### DISCUSSION TITLE
ESMValCore v2.8 release

### DISCUSSION THREAD

**Discussion by remi-kazeroni** [ESMValGroup/ESMValCore#1924] [Releases]
Hi @ESMValGroup/esmvaltool-developmentteam, the release process for ESMValCore v2.8.0 is ongoing! 

### Feature freeze
Friday, March 03. 

### Release team
Release manager: @remi-kazeroni 
Co-release manager: @valeriupredoi 

### Release branch
[`v2.8.x`](https://github.com/ESMValGroup/ESMValCore/tree/v2.8.x)

### Release Status
| Minor Release  | Status | Milestone | Date | Docs | Release Tag | PyPi | conda-forge | docker | testing | 
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| `v2.8.0rc1`  | Released  | [Core v2.8](https://github.com/ESMValGroup/ESMValCore/milestone/14) | `03/03/2023` | [Docs](https://docs.esmvaltool.org/projects/ESMValCore/en/v2.8.0rc1/) | [v2.8.0rc1](https://github.com/ESMValGroup/ESMValCore/releases/tag/v2.8.0rc1) | [PyPi/v2.8.0rc1](https://pypi.org/project/ESMValCore/2.8.0rc1/) | [anaconda](https://anaconda.org/conda-forge/esmvalcore/files) | [docker](https://hub.docker.com/layers/esmvalgroup/esmvalcore/v2.8.0rc1/images/sha256-81330d33db36f18f3d161e1226c3a4b7fc0c74818c06ff50acdae89f5d39550e?context=explore) | [webpage](https://esmvaltool.dkrz.de/shared/esmvaltool/v2.8.0rc1/) |
| `v2.8.0rc2`  | Released  | [Core v2.8](https://github.com/ESMValGroup/ESMValCore/milestone/14) | `17/03/2023` | [Docs](https://docs.esmvaltool.org/projects/ESMValCore/en/v2.8.0rc2/) | [v2.8.0rc2](https://github.com/ESMValGroup/ESMValCore/releases/tag/v2.8.0rc2) | [PyPi/v2.8.0rc2](https://pypi.org/project/ESMValCore/2.8.0rc2/) | [anaconda](https://anaconda.org/conda-forge/esmvalcore/files) | [docker](https://hub.docker.com/layers/esmvalgroup/esmvalcore/v2.8.0rc2/images/sha256-fe9c055ecabbcc561abf19f7a598bc776de3fd6714c0aa9062e182987c609058?context=explore) | [webpage](https://esmvaltool.dkrz.de/shared/esmvaltool/v2.8.0rc2/debug.html) |
| `v2.8.0`  | Released  | [Core v2.8](https://github.com/ESMValGroup/ESMValCore/milestone/14) | `23/03/2023` | [Docs](https://docs.esmvaltool.org/projects/ESMValCore/en/v2.8.0/) | [v2.8.0](https://github.com/ESMValGroup/ESMValCore/releases/tag/v2.8.0rc2) | [PyPi/v2.8.0](https://pypi.org/project/ESMValCore/2.8.0/) | [anaconda](https://anaconda.org/conda-forge/esmvalcore) | [docker](https://hub.docker.com/layers/esmvalgroup/esmvalcore/v2.8.0/images/sha256-0be0dcf7797c446e92315235fa71e9744244a36e8f36a34e3db1b748444771d3?context=explore) | [webpage](https://esmvaltool.dkrz.de/shared/esmvaltool/v2.8.0/) |

Feel free to ask general questions about the release in this discussion. I will open separate issues about the recipe testing, output comparison, selection of highlights for the Changelog, ... 

Thanks in advance for your collaboration with the release of v2.8!

**Comment by valeriupredoi** [comment]
great overview/bird's-eye :eagle: - I'll have a go through these myself now, see if I have anything that I#m trailing the release by :beer:

**Comment by valeriupredoi** [comment]
I bumped Issues #1665 and #1800 to M2.9 - their PRs are bumped too, there is simply not much time to address them, but I will test the auto feedstock PR in a different package see if all works fine there

**Comment by remi-kazeroni** [comment]
### **Update on the feature freeze for the Core:**

- Things have been delayed due to extra testing and work to merge #1609

#### Remaining PRs
(see https://github.com/ESMValGroup/ESMValCore/milestone/14)
- Now that we have #1609 in, we can focus on the last remaining PRs that were depending on it:
  - [x] https://github.com/ESMValGroup/ESMValCore/pull/1935
  - [x] https://github.com/ESMValGroup/ESMValCore/pull/1936
  - [x] https://github.com/ESMValGroup/ESMValCore/pull/1937
  - [x] https://github.com/ESMValGroup/ESMValCore/pull/1930 (if approved on time)

#### New urgent issue?

  - [x] https://github.com/ESMValGroup/ESMValCore/issues/1948

#### New Timeline
- Next Tuesday, 13:00 (GMT): Tech Lead Team Meeting to discuss pending v2.8 issues and discuss the release process
- **Next Wednesday** (March 1st, around noon): **Feature freeze for the Core**
- Around March 8th: feature freeze for the Tool. I will communicate about that in the Tool repository next Monday and go through the v2.8 milestone

**Reply by bouweandela** [reply]
It would be nice to add some overview information to the top post, as they do it in iris: https://github.com/SciTools/iris/discussions/4433, https://github.com/SciTools/iris/discussions/4522. We don't use a project, but could add a link to the milestone instead.

**Reply by remi-kazeroni** [reply]
Thanks for the suggestion! Please let me know what you think about the new [top post](https://github.com/ESMValGroup/ESMValCore/discussions/1924#discussion-4853663).

**Reply by bouweandela** [reply]
Awesome, much more clear, thanks!

**Comment by Peter9192** [comment]
I noticed after #1609 the example notebook on https://esmvaltool--1609.org.readthedocs.build/projects/ESMValCore/en/1609/notebooks/loading-and-processing-data.html still uses the old `add_ancillary` and `dataset.ancillaries`. Should this be fixed before the release?

**Reply by remi-kazeroni** [reply]
Good find! It would be great to update these notebooks. Would you or @bouweandela have the time to update these notebooks for v2.8? I think it is not critical to have that ready for the feature freeze as we don't test notebooks but it'd be great to have that for the stable release.

**Reply by bouweandela** [reply]
I updated the notebook here: #1945

**Comment by remi-kazeroni** [comment]
The round of testing with `v2.8.0rc1` does not seem to lead to too many errors, see overview [here](https://github.com/ESMValGroup/ESMValTool/issues/3076#issuecomment-1460311698). Therefore, I'm aiming at including bugfixes into a rc2 which should be the (ideally/hopefully) the _final_ release candidate. A lot of testing has happened already during the review of several PRs, reducing the testing load for this release 🍺

**Comment by remi-kazeroni** [comment]
Plans for next week would be to address/merge the issues/PR in the [milestone v2.8](https://github.com/ESMValGroup/ESMValCore/milestone/14). This includes the Changelog for v2.8 and various issues found during the first round of testing. If things go well, we could plan a second and last release candidate towards the end of next week (e.g. March 17). This would be also the time for the feature freeze of the Tool. The Tool release branch and Core rc2 would be used for one last round of recipe testing + output comparison against last release. If all goes well, we could then think of finishing the release in about 2 weeks from now 🤞 

#### Milestone v2.8 (to finish)

- [x] https://github.com/ESMValGroup/ESMValCore/issues/1951
- [x] https://github.com/ESMValGroup/ESMValCore/issues/1962
- [x] https://github.com/ESMValGroup/ESMValCore/pull/1966
- [x] https://github.com/ESMValGroup/ESMValCore/issues/1967
~~- [ ] https://github.com/ESMValGroup/ESMValCore/issues/1964 (if agreement on this)~~

**Reply by bouweandela** [reply]
Looks good @remi-kazeroni! Could you update the table at the top with the next planned release candidate?

**Comment by remi-kazeroni** [comment]
Apart from finalizing the changelog for v2.8.0 (WIP in #1959), everything is ready to proceed to the release of the second (and final) release candidate for the Core v.2.8.0. The milestone is now empty. I am planning to make the release on GitHub and PyPi tomorrow,

**Reply by remi-kazeroni** [reply]
`v2.8.0rc2` is now available on GitHub and PyPi, see top [post](https://github.com/ESMValGroup/ESMValCore/discussions/1924#discussion-4853663)

**Comment by valeriupredoi** [comment]
@remi-kazeroni you an absolute legend! I am heading over to the Changelog now

**Comment by remi-kazeroni** [comment]
Since no Core issues were noticed in the second round of testing (see https://github.com/ESMValGroup/ESMValTool/issues/3127), @valeriupredoi and myself have decided to do the release of ESMValCore v2.8 tomorrow already: GitHub release and then on conda-forge. This will give us a few days to adjust the environment in the Tool, conda-lock files, ...

**Comment by remi-kazeroni** [comment]
The release of ESMValCore v2.8.0 is now completed! The package is available on all our platforms, see [top post](https://github.com/ESMValGroup/ESMValCore/discussions/1924#discussion-4853663). Once the release of ESMValTool v2.8.0 is completed (mid-next week), the release of both packages will be announced through our communication channels. Stay tuned!

**Reply by bouweandela** [reply]
Thanks @remi-kazeroni and @valeriupredoi and the rest of the team! Great to see how well organized and clearly communicated this release was and how well you managed to engage the wider community.
